src/estimation/estimation_main.py

```diff
@@ -47,7 +47,8 @@ def run_estimation(
     weighted_df = weights.outlier_weights(df)
 
     # apply the weights to the dataframe and apply the specified rounding
-    estimated_df, num_cols = appweights.apply_weights(weighted_df, config, 4)
+    for_est = weighted_df.copy()
```

Comment:
so was that the bug, then? That we hadn't done .copy() and the orig dataframe got updated??

Reply:
Yes, that was it!